proto/layout/layout_node.proto

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package com.android.designcompose.layout.proto;
+import "layout_style.proto";
+option java_multiple_files = true;
+
+// The final result of a layout algorithm for a single taffy Node
+message Layout {
+  // Relative ordering of the node. Render nodes with a higher order on top
+  // of nodes with lower orde
+  uint32 order = 1;
+  float width = 2;
+  float height = 3;
+  float left = 4;  // The top-left corner of the node
+  float top = 5;
+}
+
+/* The layout response sent back to client which contains a layout state ID and
+a list of layout IDs that have changed.
+*/
+message LayoutChangedResponse {
+  int32 layout_state = 1;
+  map<int32, Layout> changed_layouts = 2;
+}
+
+// A representation of a Figma node to register for layout.
+message LayoutNode {
+  int32 layout_id = 1;
+  int32 parent_layout_id = 2;
+  int32 child_index = 3;
+  LayoutStyle style = 4;
+  string name = 5;
+  bool use_measure_func = 6;
+  optional int32 fixed_width = 7;
+  optional int32 fixed_height = 8;
```

Comment:
Nope, required was. Basically all `Messages` are optional, but the basic types are not optional by default. If you want to be able to specify that the field has not been set / is null and it's a basic type then you need to specify optional. Otherwise if the field were missing then it'd default to the default value for that type (0 for ints, etc.)

Here's the doc that most clearly explains it (but in a very technical way):
https://protobuf.dev/programming-guides/field_presence/#presence-in-proto3-apis